enterprise/server/util/oci/oci.go

```diff
@@ -110,3 +121,107 @@ func (c Credentials) String() string {
+	}
+
+	remoteOpts := []remote.Option{remote.WithContext(ctx)}
+	if !credentials.IsEmpty() {
+		remoteOpts = append(remoteOpts, remote.WithAuth(&authn.Basic{
+			Username: credentials.Username,
+			Password: credentials.Password,
+		}))
+	}
+
+	remoteDesc, err := remote.Get(imageRef, remoteOpts...)
+	if err != nil {
+		if t, ok := err.(*transport.Error); ok && t.StatusCode == http.StatusUnauthorized {
+			return nil, status.PermissionDeniedErrorf("could not retrieve image manifest: %s", err)
+		}
+		return nil, status.UnavailableErrorf("could not retrieve manifest from remote: %s", err)
+	}
+
+	switch remoteDesc.MediaType {
+	// This is an "image index", a meta-manifest that contains a list of
+	// {platform props, manifest hash} properties to allow client to decide
+	// which manifest they want to use based on platform.
+	case types.OCIImageIndex, types.DockerManifestList:
+		imgIdx, err := remoteDesc.ImageIndex()
+		if err != nil {
+			return nil, status.UnknownErrorf("could not get image index from descriptor: %s", err)
+		}
+		imgs, err := partial.FindImages(imgIdx, match.Platforms(v1.Platform{
+			Architecture: platform.GetArch(),
+			OS:           platform.GetOs(),
+			Variant:      platform.GetVariant(),
+		}))
+		if err != nil {
+			return nil, status.UnavailableErrorf("could not search image index: %s", err)
+		}
+		if len(imgs) == 0 {
+			return nil, status.NotFoundErrorf("could not find suitable image in image index")
+		}
+		if len(imgs) > 1 {
+			return nil, status.NotFoundErrorf("found multiple matching images in image index")
+		}
+		return imgs[0], nil
+	case types.OCIManifestSchema1, types.DockerManifestSchema2:
+		img, err := remoteDesc.Image()
+		if err != nil {
+			return nil, status.UnknownErrorf("could not get image from descriptor: %s", err)
+		}
+		return img, nil
+	default:
+		return nil, status.UnknownErrorf("descriptor has unknown media type %q", remoteDesc.MediaType)
+	}
+}
+
+func Authorized(ctx context.Context, image v1.Image, imageName string, credentials Credentials) error {
+	imageRef, err := ctrname.ParseReference(imageName)
+	if err != nil {
+		return err
+	}
+
+	// Check if we have access to all the layers.
+	layers, err := image.Layers()
+	if err != nil {
+		return status.UnknownErrorf("could not get layers for image: %s", err)
+	}
+	eg, egCtx := errgroup.WithContext(ctx)
+	remoteOpts := []remote.Option{remote.WithContext(egCtx)}
+	if !credentials.IsEmpty() {
+		remoteOpts = append(remoteOpts, remote.WithAuth(&authn.Basic{
+			Username: credentials.Username,
+			Password: credentials.Password,
+		}))
+	}
+
+	for _, layerInfo := range layers {
+		layerInfo := layerInfo
+		eg.Go(func() error {
+			d, err := layerInfo.Digest()
+			if err != nil {
+				return err
+			}
+			layerRef := imageRef.Context().Digest(d.String())
+			l, err := remote.Layer(layerRef, remoteOpts...)
+			if err != nil {
+				return err
+			}
+			// This issues a HEAD request for the layer.
+			_, err = l.Size()
+			return err
+		})
+	}
```

Comment:
I am not super familiar with this but just to double check, does podman do auth in the same way (checking access to all layers)? Or is it sufficient to just authorize access to the manifest?

Reply:
`podman pull`ing an image that's already been pulled just issues a GET request for the manifest. Given that, I went ahead and removed this function entirely because we don't need it for the re-authorization check we discussed and it's not really adding any value in sociartifactstore.go because the layers are fetched later on for indexing anyways.